modules/apiHandler.py

```diff
@@ -232,4 +234,60 @@ def getCodeForcesLastSubmissionId(self, username):
             return data
         else:
             print("Could't get submission id. Please try this problem again later.")
-            return ""
\ No newline at end of file
+            return ""
+
+    # Online
+    def getSPOJSolveData(self, username):
+        """
+            This will be used in retrive already solved problem list of user
+        """
+        start = 0
+        self.submissions = []
+        previd = -1
+        currid = 0
+
+        for i in range(1000):
```

Comment:
Can you be a little more specific about the range and iteration variable i? It's really hard to understand at first glance. Also, you are using another i inside which is a bad practice and I can see some more variable names which doesn't clarify what it's used for.

It would really help if you improved the variable naming. I'm not saying I followed the naming conventions earlier but as I now know about it's importance, I'm trying to follow that and prefer to improve readability which in terms help other contributors.

Thanks

Reply:
we are assuming an user has 1000 page of submission history page at most (a single page has 20 submission) and we are also breaking loop if we find last page before 1000. I re-declare variable with proper meaning and also comment out to understand code